### ISSUE TITLE
Question: how to calculate values for planeNormal?

### ISSUE THREAD

**Issue by eric1foard** [viromedia/viro#408]
1. Development OS: Mac
2. Device OS & Version: iOS 11.4
3. Version: ViroReact: 2.10.0 and React Native: 0.55.1
4. Device(s): iPhone 7

## Description
Hi there! Thanks to you folks for creating an incredible library here. I have a question around defining a plane for `FixedToPlane` dragging. But first a little background on my use case:

I'm developing an application that allows users to fix posters to a wall using AR. I decided to create an interaction that uses horizontal plane detection as opposed to vertical, since in my testing horizontal plane detection is much more consistent. The interaction pattern is as follows:

1) user points phone at floor to detect a horizontal plane
2) A line (actually a long, skinny ViroBox) appears when plane is detected, which can be dragged along the floor (i.e. it is fixed to the floor that was detected):
![img_1105](https://user-images.githubusercontent.com/8867620/45248057-900cf280-b2c1-11e8-8d6d-3fbf088f8ce7.PNG)
3) The user drags and rotates this line such that it is aligned with where the floor meets the wall:
![img_1107](https://user-images.githubusercontent.com/8867620/45248521-8d60cc00-b2c6-11e8-994f-a1306ecff782.PNG)
4) The image appears, and can be dragged into the desired position anywhere on the wall. Since we know the angle of the line, we know the angle at which to display the image so it appears fixed to the wall:
![img_1108](https://user-images.githubusercontent.com/8867620/45248523-9a7dbb00-b2c6-11e8-8279-d16db7adc1af.PNG)

My confusion arises with how to configure `FixedToPlane` for the wall.

When I defined the `FixedToPlane` dragging along the floor, this is straightforward since the normal vector just points directly upward (exactly as described [here](https://github.com/viromedia/virocore/issues/107#issuecomment-394854242)). Relevant code:
```
<ViroBox
            ref={c => this.state.viroNode = c}
            onDrag={pos => { currPos = pos; }}
            dragType='FixedToPlane'
            dragPlane={{
              planePoint: anchorPt,
              planeNormal: [0, 1, 0],
              maxDistance: 5
            }}
            height={0.01}
            length={0.01}
            width={5}
            visible={!showImage}
          />
```

However, for dragging along the wall, I'm unsure of how to calculate the `planeNormal` property. If the user was standing perpendicular to the wall when they performed step 3 above, then I could just do this:
```
<ViroImage
            onDrag={() => {}}
            dragType='FixedToPlane'
            dragPlane={{
              planePoint: currPos,
              planeNormal: [0, 0, 1], // z = 1 only works if user standing perpendicular to wall. What to put here?
              maxDistance: 5
            }}
            source={{ uri: image }}
            height={heightFormatted}
            width={widthFormatted}
            position={[0, 1.5, 0]}
            rotation={[0, currRotation, 0]}
            visible={showImage}
          />
```

But you might imagine that the user may be standing at an angle relative to the wall, in which case the drag plane would not be right.

TL;DR: given that I know the rotation relative to the user of the plane I'd like to create, how do I calculate the correct values for `planeNormal`?

Thanks again!

**Comment by achuvm** [comment]
Hi @eric1foard,

You can calculate this vector based on the given transform of the `ViroBox` using the `getTransformAsync()`. You'll need to do some vector transform to get the forward vector of the box based on the returned rotation of the box. 

This is because the plane you want to drag on will have the same forward vector as the box the user placed.

Let us know if that works!

**Comment by eric1foard** [comment]
@achuvm that makes so much sense! Thanks so much for your response. I will try that out and let you know!

**Comment by eric1foard** [comment]
@achuvm unfortunately my math is weak in this area so I'm relying on google to figure out how to calculate a forward vector given the rotation of the viroBox. Could you weigh in on how you would achieve this? Here's what I have so far:

```
  forwardVector() {
    const yaw = this.props.meta.rotation[0]; // comes from getTransformAsync
    const pitch = this.props.meta.rotation[1];
    return [ // array passed into dragPlane.planeNormal of ViroImage
      Math.cos(pitch) * Math.cos(yaw),
      Math.cos(pitch) * Math.sin(yaw), 
      Math.sin(pitch)                 
    ];
  }```

It seems that my above function does not calculate the plane correctly; any guidance here would be much appreciated? Thanks so much

**Comment by achuvm** [comment]
Hey @eric1foard,

One way of doing this is to compute the rotation matrix for each axis as described here: 
https://en.wikipedia.org/wiki/Rotation_matrix#Basic_rotations

Once you have the individual X, Y and Z Rotation matrices, then you'll need to multiply them together to get the Rotation matrix:

`R<sub>all</sub> = R<sub>z</sub> x R<sub>y</sub> x R<sub>x</sub>`

Then multiply `R<sub>all</sub>` by your vector to get the rotated vector.

I advise using some matrix math library such as: 
https://www.npmjs.com/package/matrixmath

**Comment by eric1foard** [comment]
Thanks @achuvm I will give that a try

**Comment by alexzaganelli** [comment]
Hi @eric1foard, may you please contact me at me [at] alexzaganelli.com

I would ask you an urgent information regarding this issue. Thank you so much in advance.

**Comment by skizzo** [comment]
Hi @eric1foard, did you manage to get this working? Trying to implement similar functionality with Viro.
